megamek/src/megamek/common/TargetRoll.java

```diff
@@ -198,79 +202,79 @@ public void append(TargetRoll other) {
     }
 
     /**
-     * Append another TargetRoll to the end of this one,
-     * possibly ignoring non-cumulative Modifier in the other
-     * one
+     * Append another TargetRoll to the end of this one, possibly discarding non-cumulative modifier
+     * in the other one.
      *
-     * @param other
-     * @param appendNonCumulative
+     * @param other the TargetRoll to append
+     * @param appendNonCumulative True to append all modifiers, false to append only cumulative modifiers
      */
     public void append(TargetRoll other, boolean appendNonCumulative) {
-        if (other == null) {
-            return;
-        }
-        for (TargetRollModifier modifier : other.modifiers) {
-            // possibly only add cumulative mods
-            if (appendNonCumulative || modifier.isCumulative()) {
-                addModifier(modifier);
+        if (other != null) {
+            for (TargetRollModifier modifier : other.modifiers) {
+                if (appendNonCumulative || modifier.isCumulative()) {
+                    addModifier(modifier);
+                }
             }
         }
     }
 
     /**
      * Remove all automatic failures or successes, but leave impossibles intact
      */
-
     public void removeAutos() {
         removeAutos(false);
     }
 
     /**
      * Remove all automatic failures or successes, and possibly also remove
-     * impossibles
+     * impossibles.
      *
-     * @param removeImpossibles <code>boolean</code> value wether or not
-     *                          impossibles should be removed
+     * @param removeImpossibles When true, IMPOSSIBLEs are also removed
      */
-
     public void removeAutos(boolean removeImpossibles) {
-        List<TargetRollModifier> toKeep = new ArrayList<>();
-        for (TargetRollModifier modifier : modifiers) {
-            if (!removeImpossibles) {
-                if ((modifier.getValue() != AUTOMATIC_FAIL)
-                        && (modifier.getValue() != AUTOMATIC_SUCCESS)) {
-                    toKeep.add(modifier);
-                }
-            } else if ((modifier.getValue() != AUTOMATIC_FAIL)
-                    && (modifier.getValue() != AUTOMATIC_SUCCESS)
-                    && (modifier.getValue() != IMPOSSIBLE)) {
-                toKeep.add(modifier);
-            }
+        if (removeImpossibles) {
+            modifiers.removeIf(this::isAutomaticOrImpossible);
+        } else {
+            modifiers.removeIf(this::isAutomatic);
         }
-        modifiers = toKeep;
         recalculate();
     }
 
+    private boolean isAutomaticOrImpossible(TargetRollModifier modifier) {
+        return modifier.getValue() == AUTOMATIC_FAIL || modifier.getValue() == AUTOMATIC_SUCCESS
+                || modifier.getValue() == IMPOSSIBLE;
```

Comment:
This can also be accomplished fairly quickly and cleanly with:

    return Set.of(AUTOMATIC_FAIL, AUTOMATIC_SUCCESS, IMPOSSIBLE).contains(modifier.getValue());